cmd/clusterctl/client/cluster/workload_cluster.go

```diff
@@ -53,7 +55,30 @@ func (p *workloadCluster) GetKubeconfig(workloadClusterName string, namespace st
 	}
 	dataBytes, err := utilkubeconfig.FromSecret(ctx, cs, obj)
 	if err != nil {
-		return "", errors.Wrapf(err, "\"%s-kubeconfig\" not found in namespace %q", workloadClusterName, namespace)
+		secretName := utilsecret.Name(workloadClusterName, utilsecret.Kubeconfig)
+		return "", errors.Wrapf(err, "%q not found in namespace %q", secretName, namespace)
```

Comment:
Let's just generate the error inside of the FromSecret func

same for FromUserSecret